pkg/stub/handler.go

```diff
@@ -368,44 +443,62 @@ func (h *Handler) CreateDefaultResourceIfNeeded() error {
 		if err != nil {
 			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
 		}
-		h.samplesResource = nil
-	} else {
-		srcfg = h.samplesResource
 	}
-	if srcfg != nil {
-		logrus.Println("SampleResource already received, not creating default")
-		return nil
-	}
-	// "4a" in the "startup" workflow, just create default
-	// resource and set up that way
-	srcfg = &v1alpha1.SamplesResource{}
-	srcfg.Name = v1alpha1.SamplesResourceName
-	srcfg.Kind = "SamplesResource"
-	srcfg.APIVersion = "samplesoperator.config.openshift.io/v1alpha1"
-	srcfg.Spec.Architectures = append(srcfg.Spec.Architectures, v1alpha1.X86Architecture)
-	srcfg.Spec.InstallType = v1alpha1.CentosSamplesDistribution
+
 	logrus.Println("creating default SamplesResource")
 	err = h.sdkwrapper.Create(srcfg)
 	if err != nil {
 		if !kerrors.IsAlreadyExists(err) {
 			return err
 		}
-		logrus.Println("got already exists error on create default, just going to forgo this time vs. retry")
-		// just return the raw error and initiate the sdk's requeue/ratelimiter setup
+		logrus.Println("got already exists error on create default")
 	}
 	h.deleteInProgress = false
-	return err
+	return nil
 }
 
-func (h *Handler) manageDockerCfgSecret(deleted bool, samplesResource *v1alpha1.SamplesResource, secret *corev1.Secret) error {
-	if secret.Name != v1alpha1.SamplesRegistryCredentials {
+func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) error {
+	if srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) {
 		return nil
 	}
 
-	if h.samplesResource == nil && h.waitingForCredential == nil {
-		if !deleted {
-			h.registrySecret = secret
+	secret, err := h.secretclientwrapper.Get(h.namespace, v1alpha1.SamplesRegistryCredentials)
+	// have to check both since get will return an empty secret struct along with not found errs
+	// handle case where secret is created in sample operator namespace before SamplesResource
+	if secret != nil && !kerrors.IsNotFound(err) {
+		return h.manageDockerCfgSecret(false, srcfg, secret)
+	}
+
+	// if trying to do rhel to the default registry.redhat.io registry requires the secret
+	// be in place since registry.redhat.io requires auth to pull; since it is not ready
+	// log error state
+	if srcfg.Spec.InstallType == v1alpha1.RHELSamplesDistribution &&
+		(srcfg.Spec.SamplesRegistry == "" || srcfg.Spec.SamplesRegistry == "registry.redhat.io") {
+
+		if !srcfg.ConditionFalse(v1alpha1.SamplesExist) {
+			err := fmt.Errorf("Cannot create rhel imagestreams to registry.redhat.io without the credentials being available: %#v", srcfg)
+			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionFalse, err, "%v")
+		} else {
+			// already recorded error/need
+			return fmt.Errorf("Previously noted that cannot create rhel imagestreams to registry.redhat.cio with the credentials being available %#v", srcfg)
```

Comment:
how do we know this was previously noted?  Seems like SamplesExist could be false for any number of reasons when we get here, not necessary just because the credentials didn't exist (even if there isn't such a reason today, there could be in the future).

Reply:
because all the h.processError call above does is set the SamplesExist condition to false

I saw no need to induce more churn on the object and update again.  Returning the error still kicks in relist, etc.

If the if check above evolves then yes this path might need to evolve with it

Reply:
I would expect us to start defaulting SamplesExist to false, or setting SamplesExist to false any time we see that the configuration is invalid(and we haven't already created the samples), at which point this logic would be flawed.  We can proceed w/ this for now but i think it's another area of behavior we'll need to revisit in the future.

Reply:
I'll update to always call with some clarifying comment about consciously not trying to minimize duplicate condition settings